libs/langchain/langchain/document_loaders/parsers/pdf.py

```diff
@@ -206,28 +206,31 @@ def lazy_parse(self, blob: Blob) -> Iterator[Document]:
         import fitz
 
         with blob.as_bytes_io() as file_path:
-            doc = fitz.open(file_path)  # open document
-
-            yield from [
-                Document(
-                    page_content=page.get_text(**self.text_kwargs)
-                    + self._extract_images_from_page(doc, page),
-                    metadata=dict(
-                        {
-                            "source": blob.source,
-                            "file_path": blob.source,
-                            "page": page.number,
-                            "total_pages": len(doc),
-                        },
-                        **{
-                            k: doc.metadata[k]
-                            for k in doc.metadata
-                            if type(doc.metadata[k]) in [str, int]
-                        },
-                    ),
-                )
-                for page in doc
-            ]
+            if isinstance(blob.data, bytes):
```

Comment:
I didn't modify pdf.py

Reply:
because when the data is coming from the memory (i.e. Blob.from_data(self.file_path.read())) fitz.open(stream) should be used and when it's coming from a file (i.e. blob = Blob.from_data(self.file_path.read())) fitz.open(filename) should be used.